databases/Assignment2_answers.sql

```diff
@@ -0,0 +1,316 @@
+
+2. Find the set of all manufacturers ordered alphabetically.
+SELECT DISTINCT(Manufacturer)
+FROM Product
+ORDER BY DESC; # is there an alpha order?
+# all data types are by default order-able
+
+
+3. Which computer models appear in the database?
+SELECT Model
+FROM Product
+WHERE Type = pc OR Type = laptop;
+
+OR --> (below is a better idea lol)
+
+SELECT DISTINCT(Model)
+FROM Computer;
+
+
+4. Which computer models use 256 MB of RAM?
+SELECT Model
+FROM Computer
+WHERE Computer.RAM = 256;
+
+5. Which computer models cost less than $1100?
+SELECT Model
+FROM Computer
+WHERE Price < 1100;
+
+6. Which laptop models have a HD of at least 50 GB and cost less than $1400?
+SELECT Model
+FROM Computer
+INNER JOIN Product USING Model
+WHERE Product.Type = "laptop" Computer.HD >= 50 AND Price < 1400;
+
+7. Which laser printer models can print color documents?
+SELECT Model
+FROM Printer
+WHERE Printer.Type = laser AND Printer.Color = true;
+
+8. Which laptop models that have more than 256 MB of RAM
+are produced by manufacturer D?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.RAM > 256 AND Product.Manufacturer = D;
+
+9. Which computer models that have more than 256 MB of RAM
+are produced by manufacturer D?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Computer.RAM > 256 AND Product.Manufacturer = D;
+
+10. Which laptop models cost less than $1200?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.Price < 1200;
+
+11. Which manufacturers produce personal computers and printers?
+SELECT DISTINCT Manufacturer
+FROM Product
+WHERE Type = pc OR Type = printer;
+
+12. Which manufacturers produce all the product types?
+SELECT Manufacturer
+FROM Product
+GROUP BY Manufacturer
+HAVING COUNT(DISTINCT Type) = 3;
+
+13. Which manufacturers supply only personal computers?
+
+SELECT distinct manufacturers
+FROM Product
+WHERE Type ='pc'
+
+
+
+with pcs AS (
+    SELECT Manufacturer, Model
+    FROM Product
+    WHERE Type = 'pc'
+),
+
+
+
+
+with not_pcs AS (
+    SELECT Manufacturer
+    FROM Product
+    WHERE Type != 'pc'
+    GROUP BY Manufacturer
+)
+SELECT Product.Manufacturer
+FROM Product
+WHERE Manufacturer NOT IN not_pcs;
+
+ALSO:
+
+select all of the ones that ONLY have one type
+intersect
+SELECT all of the ones that are only type= PC
+
+
+# sub query with Type != pc, outer left join it together ...
+
+EXCEPT; IS NOT IN
+
+HAVING SUM(Type) = 'pc';
+
+HAVING Type = 'pc';
+# HAVING MAX(TYPE) = 'pc';
+# the case where manufacturers sell PCs, but other things as well!
+
+
+# HAVING Type = pc .... i dont think this is right, but why?
+
+
+14. Which manufacturers supply only one type of product?
+SELECT Manufactuer
+FROM Product
+GROUP BY Manufactuer
+HAVING COUNT(DISTINCT Type) = 1;
+
+
+
+15. You want to buy: #think about how to intersect, instead of JOINS? which is more efficient? why?
```

Comment:
is my current answer to 14 incorrect though?
or are you just suggesting an alternative?